### PR TITLE
98 vignette on parsnip and recipes

### PR DESCRIPTION
**Pull request by ChloeYou**
Addressed feedback from last meeting.

### CODE REVIEW

File: vignettes/more-recipes.Rmd [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+library(poissonreg)
+```
+
+We will use the built-in dataset `case_death_rate_subset` and apply feature 
+engineering steps and fit a few models from `parsnip` to predict future death 
+rates by states 7 days ahead from Dec 31, 2021.
+
+```{r}
+x <- case_death_rate_subset
+x$geo_value <- as.factor(x$geo_value)
```
Comment (ChloeYou) [inline vignettes/more-recipes.Rmd:40]: Remove this. Add to lower chunk and run model with the dummy variables.

File: vignettes/more-recipes.Rmd [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+# Preprocessing and Feature Engineering
+
+In this section, we focus on showing how `step_*()` functions from the `recipes`
+package can also be applied in our `epipredict` workflows. 
+
+First, create a recipe object from the original data and then specify the 
+preprocessing steps. Steps are then sequentially added.
+
+`recipes` package also allows [manual alterations of roles](https://recipes.tidymodels.org/reference/roles.html)
```
Comment (ChloeYou) [inline vignettes/more-recipes.Rmd:55]: Explain what roles are.

File: vignettes/more-recipes.Rmd [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+The function automatically categorizes roles into `geo_value`, `time_value` and 
+`raw` based on variable names, where `raw` includes `case_rate` and `death_rate`. 
+
+We first create a made-up population dataset by states to allow the conversion
+from `case_rate` and `death_rate` into `cases` and  `deaths`. The binary variable
+`social_distancing` indicates whether the state followed social distancing rules.
+A 'throw-away' column with all ones is also created to show that `step_rm()` 
+can be used to remove unwanted columns.
+```{r madeup-population-data}
```
Comment (ChloeYou) [inline vignettes/more-recipes.Rmd:83]: Use real population data. covidcast::
state_census %>% select(POPESTIMATE2019, ABBR)

Comment (ChloeYou) [reply vignettes/more-recipes.Rmd:83]: Facebook survey has percentage of social distancing data.

File: vignettes/more-recipes.Rmd [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+preprocessing steps. Steps are then sequentially added.
+
+`recipes` package also allows [manual alterations of roles](https://recipes.tidymodels.org/reference/roles.html) 
+in bulk. There are a couple handy functions that can be used together to help us 
+manipulate variable roles easily. 
+
+`update_role()` alters an existing role in the recipe or assigns an initial role 
+to variables that do not yet have a declared role.
+
+`add_role()` adds an additional role to variables that already have a role in
```
Comment (ChloeYou) [inline vignettes/more-recipes.Rmd:62]: do this to geo_type and then convert to factor to use for dummy variables.

Comment (ChloeYou) [reply vignettes/more-recipes.Rmd:62]: use this along with add_role() on time_value and dummy geo_values to add in the model as covariates.

Show how to use formula in epi_recipe() as well.

File: vignettes/more-recipes.Rmd [hunk 5]
```diff
@@ -0,0 +1,277 @@
+---
+title: Apply other steps and models from recipes and parsnip packages
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Apply other steps and models from recipes and parsnip packages}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+The `epipredict` package utilizes the `tidymodels` framework, namely
```
Comment (ChloeYou) [inline vignettes/more-recipes.Rmd:9]: Add section: Not doing it the Epipredict for lagging and leading. Show demo of doing it without step_lag/lead() and how it results in not using information from the most x-lagged time points. whether leading or lagging, they will be off by one time-period.

File: vignettes/more-recipes.Rmd [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+p <- predict(wf, latest) %>% filter(!is.na(.pred))
+
+p
+```
+
+The predictions are saved in the `.pred` column and represents the log-transformed 
+predicted death counts 7 days from Dec 31. 2021. 
+
+
+### Poisson regression
```
Comment (ChloeYou) [inline vignettes/more-recipes.Rmd:227]: For each model, create its own section. Poisson regression first for original death count data which can be downloaded from the website. Then linear reg with population scales and introduce postprocessing and quantile- prediction first, then the quantile, then the population scaling. Don't reverse the order or bad things will happen.

File: vignettes/more-recipes.Rmd [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+
+predict(wf, latest) %>% filter(!is.na(.pred))
+```
+
+Let's also take a look at the poisson regression fit:
+```{r, echo =FALSE}
+wf$fit$fit
+```
+
+### Logistic Regression
```
Comment (ChloeYou) [inline vignettes/more-recipes.Rmd:254]: https://bccovid-19group.ca/publication/mcdonald-2021-12-13-auxiliary/

Use the hotspot prediction model. Three categories, upswing, flat, decrease.

**Review by ChloeYou — COMMENTED** [review]
Add to-dos.

File: vignettes/more-recipes.Rmd (hunk 7 above)
Comment (ChloeYou) [reply vignettes/more-recipes.Rmd:254]: still need to add mathematical expression

**Comment by dajmcdon** [comment]
@ChloeYou The framework you have here is really great. I made a few updates, most pretty minor. Though I also had to make a bunch of changes to some code.
1. Can you read over it all?  
2. Take a look at the changes to `layer_population_scaling()`. I added an argument to deal with "cases per 100K population". Basically `case_rate` isn't per capita, but per 100K inhabitants. Can you proof the documentation changes I made?
3. Can you make a similar modification to `step_population_scaling()`.
4. Check over the documentation for `state_census`. I decided to add it here rather than grabbing from `covidcast`.

Thank you!

**Review by dajmcdon — APPROVED** [review]

**Comment by ChloeYou** [comment]
> @ChloeYou The framework you have here is really great. I made a few updates, most pretty minor. Though I also had to make a bunch of changes to some code.
> 
> 1. Can you read over it all?
> 2. Take a look at the changes to `layer_population_scaling()`. I added an argument to deal with "cases per 100K population". Basically `case_rate` isn't per capita, but per 100K inhabitants. Can you proof the documentation changes I made?
> 3. Can you make a similar modification to `step_population_scaling()`.
> 4. Check over the documentation for `state_census`. I decided to add it here rather than grabbing from `covidcast`.
> 
> Thank you!

Thank you for reviewing and making edits! I'll aim to get these done by EoW. Thanks!

**Comment by ChloeYou** [comment]
> @ChloeYou The framework you have here is really great. I made a few updates, most pretty minor. Though I also had to make a bunch of changes to some code.
> 
> 1. Can you read over it all?
> 2. Take a look at the changes to `layer_population_scaling()`. I added an argument to deal with "cases per 100K population". Basically `case_rate` isn't per capita, but per 100K inhabitants. Can you proof the documentation changes I made?
> 3. Can you make a similar modification to `step_population_scaling()`.
> 4. Check over the documentation for `state_census`. I decided to add it here rather than grabbing from `covidcast`.
> 
> Thank you!

On the second point, `proof the documentation change` as in write a test case for the functionality?

**Comment by ChloeYou** [comment]
Created new PR in https://github.com/cmu-delphi/epipredict/pull/137 to address remaining comments. @dajmcdon

**Pull request merged**